lms/static/js/student_account/views/account_settings_fields.js

```diff
@@ -1,191 +1,233 @@
+
+                    var view = this;
+                    $.ajax({
+                        type: 'POST',
+                        url: '/i18n/setlang/',
+                        data: data,
+                        dataType: 'html',
+                        success: function () {
+                            view.showSuccessMessage();
+                        },
+                        error: function () {
+                            view.showNotificationMessage(
+                                view.indicators.error +
+                                gettext('You must sign out and sign back in before your language changes take effect.')
+                            );
+                        }
+                    });
                 }
-            }
-        });
-
-        AccountSettingsFieldViews.AuthFieldView = FieldViews.LinkFieldView.extend({
 
-            initialize: function (options) {
-                this.options = _.extend({}, options);
-                this._super(options);
-                _.bindAll(this, 'redirect_to', 'disconnect', 'successMessage', 'inProgressMessage');
-            },
-
-            render: function () {
-                var linkTitle;
-                if (this.options.connected) {
-                    linkTitle = gettext('Unlink');
-                } else if (this.options.acceptsLogins) {
-                    linkTitle = gettext('Link')
-                } else {
-                    linkTitle = ''
+            }),
+            PasswordFieldView: FieldViews.LinkFieldView.extend({
+                fieldType: 'button',
+                fieldTemplate: field_link_account_template,
+                events: {
+                    'click button': 'linkClicked'
+                },
+                initialize: function (options) {
+                    this.options = _.extend({}, options);
+                    this._super(options);
+                    _.bindAll(this, 'resetPassword');
+                },
+                linkClicked: function (event) {
+                    event.preventDefault();
+                    this.resetPassword(event);
+                },
+                resetPassword: function () {
+                    var data = {};
+                    data[this.options.emailAttribute] = this.model.get(this.options.emailAttribute);
+
+                    var view = this;
+                    $.ajax({
+                        type: 'POST',
+                        url: view.options.linkHref,
+                        data: data,
+                        success: function () {
+                            view.showSuccessMessage();
+                        },
+                        error: function (xhr) {
+                            view.showErrorMessage(xhr);
+                        }
+                    });
+                },
+                successMessage: function () {
+                    return this.indicators.success + window.interpolate_text(
+                            gettext(
```

Comment:
@waheedahmed This is another message that should not be split across multiple lines. Note that we are also in favor of using double quotes for strings if they include single quotes (although that may fail the JSHint linter).